File: DataStructures/LinkedList/swapNode.java
class Node{
    public:
    int data;
    Node next;
    Node(int d)
    {
        data = d;
        next = null;
    }
}

class LinkedList{

    Node head = null;

    public void swapNodes(Node head, int x, int y)
    {
        if(x == y){
            return;
        }

        Node currX = head;
        Node prevX = null;
        while(currX && currX.data != x)
        {
            prevX = currX;
            currX = currX.next;
        }

        Node prevY = null;
        Node currY = head;

        while(currY && currY.data != y)
        {
            prevY = currY;
            currY = currY.next;
        }

        if(currX == null || currY == null)
        {
            return;
        }

        if(prevX != null)
        {
            prevX->next = currY;
        }else{
            head = currY;
        }

        if(prevY != null)
        {
            prevY->next = currX;
        }else{
            head = currX;
        }

        Node temp = currY.next;
        Node currY.next = currX.next;
        currX.next = temp;
    }

    public void push(Node head, int x)
    {
        Node new_node = Node(x);
        new_node.next = head;
        head = new_node;
    }

    public void printList(Node head)
    {
        while(head != null){
            System.out.print(head.data + " ");
            head = head.next;
        }
    }

    public static void main(String args[])
    {
        Node head = null;
        push(head, 7);
        push(head, 72);
        push(head, 3);
        push(head, 21);
        push(head, 24);
        push(head, 20);

        // we do the rest here call functions etc
    }

}

File: DataStructures/LinkedList/swapNode.c
#include <stdlib.h>
#include <stdio.h>

struct Node {
    int data;
    struct * next;
    Node(int d)
    {
        data = d;
    }
}

void swapNodes(struct ** head_ref, int x, int y)
{
    if(x == y){
        return;
    }

    struct Node * currX = * head_ref;
    struct Node * prevX = NULL;
    while(currX && currX->data != x)
    {
        prevX = currX;
        currX = currX->next;
    }

    struct Node * currY = * head_ref;
    struct Node * prevY = NULL:

    while(currY && currY ->data != y)
    {
        prevY = currY;
        currY = currY ->next;
    }

    if(currX == NULL || currY == NULL)
    {
        return;
    }

    //swapping nodes 
    if(prevX != NULL)
    {
        prevX->next = currY;
    }else{
        *head_ref = currY;
    }

    if(prevY != NULL)
    {
        prevY->next = currX;
    }else{
        *head_ref = currX;
    }

    struct Node * temp = currY->next;
    currY->next = currX -> next;
    currX->next = temp; 
}

void push(struct ** head_ref, int newData)
{
    struct Node * new_node = (struct Node*)malloc(sizeof(struct Node));
    new_node->data = newData;
    new_node->next = *head_ref;
    *head_ref = new_node;

}

void printList(struct Node node)
{
    while(node != NULL)
    {
        printf(node->data + " ");
        node = node->next;
    }
}

int main()
{
    struct Node * head = NULL:
    push(&head, 70);
    push(&head, 27);
    push(&head, 74);
    push(&head, 17);
    push(&head, 72);
    push(&head, 21);

    printf("LinkedList before swapping");

    printList(head);

    swapNodes(&head, 27, 17);

    printf("List after swapping");

    printList(head);
}